src/element/annotations.rs

```diff
@@ -0,0 +1,124 @@
+    // else (`thinvec`?) in the future. Users are unlikely to need to construct an `Annotations`
+    // themselves, but can use a `From` impl if necessary.
+    pub(crate) fn new(symbols: Vec<Symbol>) -> Self {
+        Annotations { symbols }
+    }
+
+    /// Constructs an Annotations object representing an empty symbol sequence
+    pub fn empty() -> Self {
+        Annotations {
+            symbols: Vec::new(),
+        }
+    }
+
+    /// Returns a [SymbolsIterator] that yields each of the [Symbol]s in this annotations sequence
+    /// in order.
+    pub fn iter(&self) -> SymbolsIterator {
+        self.into_iter()
+    }
+
+    /// Returns the number of annotations in this sequence.
+    /// ```
+    /// use ion_rs::element::Annotations;
+    /// let annotations: Annotations = ["foo", "bar", "baz"].into();
+    /// assert_eq!(annotations.len(), 3);
+    /// ```
+    pub fn len(&self) -> usize {
+        self.symbols.len()
+    }
+
+    /// Returns `true` if this sequence contains zero annotations. Otherwise, returns `false`.
+    /// ```
+    /// use ion_rs::element::Annotations;
+    /// let annotations: Annotations = ["foo", "bar", "baz"].into();
+    /// assert!(!annotations.is_empty());
+    /// ```
+    pub fn is_empty(&self) -> bool {
+        self.symbols.len() == 0
+    }
+
+    /// Returns `true` if any symbol in this annotations sequence is equal to the provided text.
+    /// Otherwise, returns `false`.
+    /// ```
+    /// use ion_rs::element::Annotations;
+    /// let annotations: Annotations = ["foo", "bar", "baz"].into();
+    /// assert!(annotations.contains("foo"));
+    /// assert!(annotations.contains("bar"));
+    /// assert!(annotations.contains("baz"));
+    ///
+    /// assert!(!annotations.contains("quux"));
+    /// assert!(!annotations.contains("quuz"));
+    /// ```
+    pub fn contains<S: AsRef<str>>(&self, query: S) -> bool {
+        let query: &str = query.as_ref();
+        self.iter().any(|symbol| symbol.text() == Some(query))
+    }
+
+    /// Returns the text of the first annotation in this sequence.
+    ///
+    /// If the sequence is empty, returns `None`.
+    /// If the first annotation in the sequence is `$0` (symbol ID 0), returns `None`.
+    /// Otherwise, returns a `Some(&str)` containing the text.
+    ///
+    /// To view the first annotation as a [Symbol] rather than a `&str`, use
+    /// `annotations.iter().next()`.
+    /// ```
+    /// use ion_rs::element::Annotations;
+    /// use ion_rs::Symbol;
+    /// let annotations: Annotations = ["foo", "bar", "baz"].into();
+    /// assert_eq!(annotations.first(), Some("foo"));
+    ///
+    /// let empty_sequence: Vec<&str> = vec![];
+    /// let annotations: Annotations = empty_sequence.into();
+    /// assert_eq!(annotations.first(), None);
+    ///
+    /// let annotations: Annotations = [Symbol::unknown_text()].into();
+    /// assert_eq!(annotations.first(), None)
+    /// ```
+    pub fn first(&self) -> Option<&str> {
```

Comment:
I'm not thrilled with the fact that this treats "no annotations" and "the first annotation has unknown text" as equivalent, but I'm okay with it because it is explicitly called out in the documentation.

Would it be terrible to use `Result` instead of `Option` for this? Then we could distinguish them.
```rust
enum GetAnnotationError {
    NoSuchElement,
    UnknownSymbolText,
}

pub fn first(&self) -> Result<&str, AnnotationTextError> {
    // ...
}
```
Let me know what you think. I think my suggestion is an improvement (otherwise I wouldn't suggest it), but I'm willing to approve the PR as is if you disagree.

Reply:
I agree that it's more correct, but I think it only benefits those of us writing spec compliance tests. (i.e. Those of us using `$0`.) I think I'd prefer to keep it as-is and introduce a `first_symbol` or similar that returns `&Symbol` if we ever have demand for it.